src/public/pages/examples/path-params.js

```
@@ -0,0 +1,42 @@
+import K, * as U from "karet.util"
+import * as L    from "partial.lenses"
+import React     from "karet"
+
+import { PathInput }       from "../../components/restricted-input"
+import { PrettyStringify } from "../../components/pretty-stringify"
+
+const getPagePathRoot =
+  U.pipe( x => U.match( /(^\/[^/]+\/?).*/, x )[ 1 ] )
+
+const subPathGetter = pagePathRoot =>
+  U.pipe( U.replace( pagePathRoot, '' )
+                   , decodeURIComponent
+                   )
+
+const subPathSetter = pagePathRoot =>
+  U.pipe( U.map( U.when( x => x !== '/'
+                       , encodeURIComponent
+                       )
+               )
+        , U.join( '' )
+        , U.concat( pagePathRoot )
+        , U.replace( pagePathRoot + '/', pagePathRoot )
+        )
+
+const _subPathL = pagePathRoot =>
+  L.iso( subPathGetter( pagePathRoot )
+       , subPathSetter( pagePathRoot )
+       )
+const subPathL = U.lift( _subPathL )
+
+const decodeProps =
+  U.mapObjIndexed( decodeURIComponent )
+
+export const PathParams = ( { props, path } ) =>
+  <div>
+    <PathInput
+      type="text"
+      label="Path"
+      value={ U.view( subPathL( getPagePathRoot( path ) ), path ) }/>
+    <PrettyStringify space="2" obj={ decodeProps( props ) } />
```

Note that you can write `space={2}` to pass number `2` instead of string `"2"` as a prop.

---

Yes, I've variously passed in `"2"` and `{ 2 }`. I thought it might make sense to accept either one?

---

Hmm... Usually I try to avoid implicit conversions like that, but this is definitely  case where the implicit conversion is unlikely to cause major problems.  One thing that implicit conversions like that can hide is bugs where you are just passing something completely wrong to the component like accidentally passing the wrong variable from scope. What would happen then is that it is converted to NaN and then treated as default and that can hide the bug (potentially for a long time).

---

I completely agree. Implicit conversions are an absolute rarity in my code. I thought this might be an OK place to relax that rule, but on second thought, it's probably better to stick to principles in code written for a boilerplate / examples project. Don't want to give an implicit green light to bad practices.

I've gone back to the following:
```js
export default ( { value, replacer = null, space = 2 } ) =>
  <pre className="pretty-stringify">
    { U.lift( JSON.stringify )( value, replacer, space ) }
  </pre>
```
It now works with (and only with) `undefined`, `numericConstant` or `U.atom( numericConstant )`.